crates/runtime/src/databackend/postgres.rs

```diff
@@ -141,6 +143,27 @@ impl PostgresBackend {
     }
 }
 
+fn read_pg_config(
+    params: Arc<Option<HashMap<String, String>>>,
+    secrets: Option<Secret>,
+) -> Arc<Option<HashMap<String, String>>> {
+    let Some(params_val) = params.as_ref() else {
+        return params;
+    };
+    let mut new_params = params_val.clone();
+    let Some(secrets) = secrets else {
+        return params;
+    };
+    let Some(pg_pass_val) = new_params.get("pg_pass_key") else {
```

Comment:
Just use params here, we don't need to use new_params until we're actually modifying the data